lighthouse/tests/validator_client.rs

```diff
@@ -0,0 +1,337 @@
+    fn run_with_no_datadir(&mut self) -> CompletedTest {
+        // Setup temp directories
+        let tmp_dir = TempDir::new().expect("Unable to create temporary directory");
+        let tmp_path: PathBuf = tmp_dir.path().join(CONFIG_NAME);
+
+        // Add --dump-config <temp_path> to cmd.
+        self.cmd
+            .arg(format!("--{}", DUMP_CONFIG_CMD))
+            .arg(tmp_path.as_os_str());
+
+        // Run the command.
+        let _output = output_result(&mut self.cmd).expect("Unable to run command");
+
+        // Grab the config.
+        let config: Config =
+            from_reader(File::open(tmp_path).expect("Unable to open dumped config"))
+                .expect("Unable to deserialize to ClientConfig");
+        CompletedTest {
+            config,
+            dir: tmp_dir,
+        }
+    }
+}
+struct CompletedTest {
+    config: Config,
+    dir: TempDir,
+}
+impl CompletedTest {
+    fn with_config<F: Fn(&Config)>(self, func: F) {
+        func(&self.config);
+    }
+    fn with_config_and_dir<F: Fn(&Config, &TempDir)>(self, func: F) {
+        func(&self.config, &self.dir);
+    }
+}
+
+#[test]
+fn datadir_flag() {
+    CommandLineTest::new()
+        .run()
+        .with_config_and_dir(|config, dir| {
+            assert_eq!(config.validator_dir, dir.path().join("validators"));
+            assert_eq!(config.secrets_dir, dir.path().join("secrets"));
+        });
+}
+
+#[test]
+fn validators_and_secrets_dir_flags() {
+    let dir = TempDir::new().expect("Unable to create temporary directory");
+    CommandLineTest::new()
+        .flag("validators-dir", dir.path().join("validators").to_str())
+        .flag("secrets-dir", dir.path().join("secrets").to_str())
+        .run_with_no_datadir()
+        .with_config(|config| {
+            assert_eq!(config.validator_dir, dir.path().join("validators"));
+            assert_eq!(config.secrets_dir, dir.path().join("secrets"));
+        });
+}
+
+#[test]
+fn beacon_nodes_flag() {
+    CommandLineTest::new()
+        .flag("beacon-nodes", Some("localhost:1001,localhost:1010"))
+        .run()
+        .with_config(|config| {
+            assert_eq!(
+                config.beacon_nodes,
+                vec!["localhost:1001", "localhost:1010"]
+            )
+        });
+}
+
+#[test]
+fn allow_unsynced_flag() {
+    CommandLineTest::new()
+        .flag("allow-unsynced", None)
+        .run()
+        .with_config(|config| assert!(config.allow_unsynced_beacon_node));
+}
+
+#[test]
+fn disable_auto_discover_flag() {
+    CommandLineTest::new()
+        .flag("disable-auto-discover", None)
+        .run()
+        .with_config(|config| assert!(config.disable_auto_discover));
+}
+
+#[test]
+fn init_slashing_protections_flag() {
+    CommandLineTest::new()
+        .flag("init-slashing-protection", None)
+        .run()
+        .with_config(|config| assert!(config.init_slashing_protection));
+}
+
+// Tests for Graffiti flags.
+#[test]
+fn grafitti_flag() {
```

Comment:
I always get this spelling wrong too :sweat_smile: 

```suggestion
fn graffiti_flag() {
```

There's a few other occurrences in this file.